pkg/lint/lintersdb/custom_linters.go

```diff
@@ -0,0 +1,131 @@
+	"github.com/golangci/golangci-lint/pkg/lint/linter"
+	"github.com/golangci/golangci-lint/pkg/logutils"
+	"github.com/golangci/golangci-lint/pkg/report"
+)
+
+type AnalyzerPlugin interface {
+	GetAnalyzers() []*analysis.Analyzer
+}
+
+// WithCustomLinters loads private linters that are specified in the golangci config file.
+func (m *Manager) WithCustomLinters() *Manager {
+	if m.log == nil {
+		m.log = report.NewLogWrapper(logutils.NewStderrLog(logutils.DebugKeyEmpty), &report.Data{})
+	}
+
+	if m.cfg == nil {
+		return m
+	}
+
+	for name, settings := range m.cfg.LintersSettings.Custom {
+		lc, err := m.loadCustomLinterConfig(name, settings)
+
+		if err != nil {
+			m.log.Errorf("Unable to load custom analyzer %s:%s, %v", name, settings.Path, err)
+		} else {
+			m.nameToLCs[name] = append(m.nameToLCs[name], lc)
+		}
+	}
+
+	return m
+}
+
+// loadCustomLinterConfig loads the configuration of private linters.
+// Private linters are dynamically loaded from .so plugin files.
+func (m *Manager) loadCustomLinterConfig(name string, settings config.CustomLinterSettings) (*linter.Config, error) {
+	analyzers, err := m.getAnalyzerPlugin(settings.Path, settings.Settings)
+	if err != nil {
+		return nil, err
+	}
+
+	m.log.Infof("Loaded %s: %s", settings.Path, name)
+
+	customLinter := goanalysis.NewLinter(name, settings.Description, analyzers, nil).
+		WithLoadMode(goanalysis.LoadModeTypesInfo)
+
+	linterConfig := linter.NewConfig(customLinter)
+	linterConfig.EnabledByDefault = true
+	linterConfig.IsSlow = false
+	linterConfig.WithURL(settings.OriginalURL)
+
+	return linterConfig, nil
+}
+
+// getAnalyzerPlugin loads a private linter as specified in the config file,
+// loads the plugin from a .so file,
+// and returns the 'AnalyzerPlugin' interface implemented by the private plugin.
+// An error is returned if the private linter cannot be loaded
+// or the linter does not implement the AnalyzerPlugin interface.
+func (m *Manager) getAnalyzerPlugin(path string, settings any) ([]*analysis.Analyzer, error) {
+	if !filepath.IsAbs(path) {
+		// resolve non-absolute paths relative to config file's directory
+		configFilePath := viper.ConfigFileUsed()
+		absConfigFilePath, err := filepath.Abs(configFilePath)
+		if err != nil {
+			return nil, fmt.Errorf("could not get absolute representation of config file path %q: %v", configFilePath, err)
+		}
+		path = filepath.Join(filepath.Dir(absConfigFilePath), path)
+	}
+
+	plug, err := plugin.Open(path)
+	if err != nil {
+		return nil, err
+	}
+
+	analyzers, err := m.lookupPlugin(plug, settings)
+	if err != nil {
+		return nil, fmt.Errorf("lookup plugin %s: %w", path, err)
+	}
+
+	return analyzers, nil
+}
+
+func (m *Manager) lookupPlugin(plug *plugin.Plugin, settings any) ([]*analysis.Analyzer, error) {
+	symbol, err := plug.Lookup("New")
+	if err != nil {
+		analyzers, errP := m.lookupAnalyzerPlugin(plug)
+		if err != nil {
+			// TODO(ldez): use `errors.Join` when we will upgrade to go1.20.
```

Comment:
We can use `multierr.Append` as in https://github.com/golangci/golangci-lint/blob/98b95a99899914748a99d728c3c81b5ab5910110/pkg/lint/runner.go#L217

Reply:
in this context, it's a bit useless because we don't need the error types.